docs/how-tos/sensor-module.md

```diff
@@ -0,0 +1,414 @@
+1. [Generate boilerplate module code](#generate-boilerplate-module-code)
+1. [Implement the sensor API](#implement-the-sensor-api)
+1. [Make the module executable](#make-the-module-executable)
+1. [Test your module locally](#test-your-module-locally)
+1. [Upload your module](#upload-your-module-to-the-registry)
+
+{{% /alert %}}
+
+## Start with a test script
+
+Start by getting a test script working so you can check that the sensor code itself works before packaging it into a module.
+
+Since this how-to uses Python, you need a Python test script so that you can more easily wrap it in a Python-based module.
+You'll still be able to use any of Viam's SDKs with the module.
+
+This step depends completely on your sensor hardware (or software)--just find or write some script that gets readings from the sensor and prints them out.
+
+{{< expand "An example of getting air quality data from an online source (Open-Meteo)" >}}
+This example uses the Open-Meto API to get air quality data from [open-meteo.com](https://open-meteo.com/en/docs/air-quality-api#current=pm10,pm2_5&hourly=).
+The following test script is from the linked code sample generator.
+
+```python {class="line-numbers linkable-line-numbers"}
+# test-script.py
+import openmeteo_requests
+import requests_cache
+from retry_requests import retry
+
+# Set up the Open-Meteo API client with cache and retry on error
+cache_session = requests_cache.CachedSession('.cache', expire_after=3600)
+retry_session = retry(cache_session, retries=5, backoff_factor=0.2)
+openmeteo = openmeteo_requests.Client(session=retry_session)
+
+# Make sure all required weather variables are listed here
+# The order of variables in hourly or daily is important
+# to assign them correctly below
+url = "https://air-quality-api.open-meteo.com/v1/air-quality"
+params = {
+  "latitude": 44.0582,
+  "longitude": -121.3153,
+  "current": ["pm10", "pm2_5"],
+  "timezone": "America/Los_Angeles"
+}
+responses = openmeteo.weather_api(url, params=params)
+
+# Process first location.
+response = responses[0]
+print(f"Coordinates {response.Latitude()}°N {response.Longitude()}°E")
+print(f"Elevation {response.Elevation()} m asl")
+print(f"Timezone {response.Timezone()} {response.TimezoneAbbreviation()}")
+print(f"Timezone difference to GMT+0 {response.UtcOffsetSeconds()} s")
+
+# Current values. The order of variables needs to be the same as requested.
+current = response.Current()
+current_pm10 = current.Variables(0).Value()
+current_pm2_5 = current.Variables(1).Value()
+
+print(f"Current time {current.Time()}")
+print(f"Current pm10 {current_pm10}")
+print(f"Current pm2_5 {current_pm2_5}")
+```
+
+{{< /expand >}}
+
+{{< expand "An example of getting PM2.5 and PM10 readings over serial from sensor hardware" >}}
+This script is based on the [pms7003 module](https://app.viam.com/module/joyce/pms7003).
+Before creating the module, the following script could have been used to verify the code that gets the readings from the sensor.
```

Comment:
It can still be used so let's instead point out that this is something they can do to explore
```suggestion
For most modules, the implementation code is linked from the module page in the registry, so you can see how it works.
```

Reply:
This is not the code that is in the repo. This is a modified version of it, that prints things instead of logging them, and puts the main functionality directly into main() so that the script can be run independently.

Reply:
However, I have not actually tested this script since I do not have this hardware. I could ax this example and include only the first one (which I wrote and tested).

Reply:
looks like it should run to me, I think that's an ok risk to take.
I think it's still true to say the script is based on that module? You did copy the code from there. And it is useful to look at how people write their modules.

Reply:
Yeah, the sentence before this one still says that. I agree that is useful, but probably more in the implementation step. I'll tweak line 335 to be more explicit about that.

Reply:
That works. I've added a suggestion there. I'd still like this sentence gone.
```suggestion
```

Reply:
Why did you resolve this? Sentence is still there.

Reply:
Didn't mean to--may have gotten auto-resolved during rebase??
I'd also been confused because I added the sentence after line 335 to this effect and hadn't realized you'd wanted this deleted and your specific wording used. Committing now.